### PR TITLE
boards: arm: black_f407ve: Add support for on-board SPI flash

### PR DESCRIPTION
**Pull request by bsteinsbo**
board: black_f407 Add support for on-board SPI flash Winbond W25Q16

### CODE REVIEW

File: boards/arm/black_f407_generic/black_f407ve_defconfig [hunk 1]
```diff
@@ -24,6 +24,9 @@ CONFIG_CLOCK_CONTROL=y
 # Enable GPIO
 CONFIG_GPIO=y
 
+# Enable SPI
+CONFIG_SPI=y
```
Comment (erwango) [inline boards/arm/black_f407_generic/black_f407ve_defconfig:28]: It is up to application to control the activation of peripherals (aside from basic ones: clocks, gpio, serial for console) 
This should be done on sample side (`samples/drivers/spi_flash/prj.cnf`)

File: samples/drivers/spi_flash/boards/black_f407ve.conf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#
```
Comment (erwango) [inline samples/drivers/spi_flash/boards/black_f407ve.conf:1]: Remove this line

File: samples/drivers/spi_flash/boards/black_f407ve.conf [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+# Copyright (c) 2019 Peter Bigot Consulting, LLC
+# Copyright (c) 2020 Bjarne Steinsbo
+#
```
Comment (erwango) [inline samples/drivers/spi_flash/boards/black_f407ve.conf:4]: This line should be blank

File: samples/drivers/spi_flash/boards/black_f407ve.conf [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
```
Comment (erwango) [inline samples/drivers/spi_flash/boards/black_f407ve.conf:6]: Remove this line

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/black_f407_generic/black_f407ve_defconfig (hunk 1 above)
Comment (bsteinsbo) [reply boards/arm/black_f407_generic/black_f407ve_defconfig]: OK

File: samples/drivers/spi_flash/boards/black_f407ve.conf (hunk 2 above)
Comment (bsteinsbo) [reply samples/drivers/spi_flash/boards/black_f407ve.conf]: OK

File: samples/drivers/spi_flash/boards/black_f407ve.conf (hunk 3 above)
Comment (bsteinsbo) [reply samples/drivers/spi_flash/boards/black_f407ve.conf]: OK

**Review by erwango — APPROVED** [review]

File: samples/drivers/spi_flash/boards/black_f407ve.conf (hunk 4 above)
Comment (bsteinsbo) [reply samples/drivers/spi_flash/boards/black_f407ve.conf]: OK.  And thanks for the review!

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**